### PR TITLE
IN: Notification Sockets

### LINKED ISSUE #33: IN: Notification sockets
Integrate sockets so that notifications are delivered to the user immediately if they are currently online

### PR DESCRIPTION
**Pull request by trevorjr31**
### What this PR does (required):
- Add notification sockets
- Notifications are now updated in real time
Addresses issue 33
#33 

### Screenshots / Videos (front-end only):
https://user-images.githubusercontent.com/77120652/143491034-c55b6541-8fce-484f-a4c8-d78a1aa23fc5.mp4

### Any information needed to test this feature (required):
- Update a current booking at the owner will receive a live notification
(In the video the sitter and owner are the same user for simplicity sake. The 2nd incognito browser illustrates the socket will indeed work, even in this scenario.)

### Any issues with the current functionality (optional):
- N/A

### CODE REVIEW

File: server/app.js [hunk 1]
```diff
@@ -50,9 +50,12 @@ io.use((socket, next) => {
 
 io.on("connection", (socket) => {
   console.log("connected");
-
-  socket.on("goOnline", (connectionId) => {
-    console.log(`New socket connection, id:'${connectionId}'`);
+  socket.on("goOnline", (onlineUser) => {
+    console.log(`New socket connection, id:'${onlineUser.id}'`);
+    socket.join(`${onlineUser.id}`);
+  });
```
Comment (ezeilo-su) [inline server/app.js:56]: Won't there be a need to keep track of online users in this listener?

**Review by sundayezeilo — COMMENTED** [review]
Looks good!

File: server/app.js [hunk 2]
```diff
@@ -50,9 +50,12 @@ io.use((socket, next) => {
...
+  socket.on("sendNotification", (userId) => {
+    io.sockets.to(`${userId}`).emit("newNotification");
```
Comment (rajivtitus) [inline server/app.js:58]: 👍🏽

**Review by rajivtitus — COMMENTED** [review]
Great work here 🚀

File: server/app.js (hunk 2 above)
Comment (trevorjr31) [reply server/app.js:58]: 😃

**Pull request merged**